omnigibson/simulator.py

```diff
@@ -477,46 +478,50 @@ def import_object(self, obj, register=True):
             # Lastly, additionally add this object automatically to be initialized as soon as another simulator step occurs
             self.initialize_object_on_next_sim_step(obj=obj)
 
-        def remove_object(self, obj):
+        def remove_object(self, obj, has_registered=True):
             """
             Remove one or a list of non-robot object from the simulator.
 
             Args:
                 obj (BaseObject or Iterable[BaseObject]): one or a list of non-robot objects to remove
+                has_registered (bool): whether the object has been registered in the scene registry
             """
-            state = self.dump_state()
-
             objs = [obj] if isinstance(obj, BaseObject) else obj
 
-            # Omniverse has a strange bug where if GPU dynamics is on and the object to remove is in contact with
-            # with another object (in some specific configuration only, not always), the simulator crashes. Therefore,
-            # we first move the object to a safe location, then remove it.
-            pos = list(m.OBJECT_GRAVEYARD_POS)
-            for ob in objs:
-                ob.set_position_orientation(pos, [0, 0, 0, 1])
-                pos[0] += max(ob.aabb_extent)
+            if self.is_playing():
+                state = self.dump_state()
+
+                # Omniverse has a strange bug where if GPU dynamics is on and the object to remove is in contact with
+                # with another object (in some specific configuration only, not always), the simulator crashes. Therefore,
+                # we first move the object to a safe location, then remove it.
+                pos = list(m.OBJECT_GRAVEYARD_POS)
+                for ob in objs:
+                    ob.set_position_orientation(pos, [0, 0, 0, 1])
+                    pos[0] += max(ob.aabb_extent)
 
-            # One timestep will elapse
-            self.app.update()
+                # One timestep will elapse
+                self.app.update()
 
             for ob in objs:
-                self._remove_object(ob)
+                self._remove_object(ob, has_registered=has_registered)
 
-            # Update all handles that are now broken because objects have changed
-            self.update_handles()
+            if self.is_playing():
+                # Update all handles that are now broken because objects have changed
+                self.update_handles()
 
-            # Load the state back
-            self.load_state(state)
+                # Load the state back
+                self.load_state(state)
```

Comment:
After some recent work with Hang I'm under the impression that it's actually possible with physx to update the simulation without taking a time step (e.g. update collisions etc. but don't apply any impulse or displacement). That way we can avoid the step-and-load. Tracking now at #604

Reply:
Sounds good. I have always been trying to do this for the last years and didn't manage to find a way :(